src/background/api/index.js

```diff
@@ -1,19 +1,77 @@
-import * as oasis from '@oasisprotocol/client';
+import * as oasis from "@oasisprotocol/client";
 import { cointypes, TX_LIST_LENGTH } from "../../../config";
-import { amountDecimals, isNumber } from '../../utils/utils';
+import { amountDecimals, isNumber } from "../../utils/utils";
 import { commonFetch, getOasisClient } from "./request";
-import * as oasisRT from '@oasisprotocol/client-rt';
-const MAX_LENGTH = 500
+import request from "./request"; // to make commonFetch jest.spyOn'able
+import rpc from "./rpc"; // to make getRpcAccount jest.spyOn'able
+import * as oasisRT from "@oasisprotocol/client-rt";
+import getRpcAccount from "./rpc";
+const MAX_LENGTH = 500;
 /**
  * get account balance
  */
 export async function getBalance(address) {
-  let url = "/chain/account/info/" + address
-  let account = await commonFetch(url).catch(() => { })
+  let url = "/chain/account/info/" + address;
+  let account = await request.commonFetch(url).catch(() => {});
+  console.log(account);
   if (account && account.code === 0) {
-    return { ...account.data, address }
+    return { ...account.data, address };
   } else {
-    return {}
+    let account = await rpc.getRpcAccount(address);
+    const balanceAmount = account?.general?.balance || 0;
+    let balance = balanceAmount;
+    if (balance) {
+      balance = oasis.quantity.toBigInt(balance).toString();
+    }
+
+    const oasisClient = getOasisClient();
+    let shortKey = await oasis.staking.addressFromBech32(address);
+    let height = oasis.consensus.HEIGHT_LATEST;
+    let delegation = await oasisClient
+      .stakingDelegationInfosFor({ height: height, owner: shortKey })
+      .catch((err) => err);
+
+    const escrowTotal = Array.from(delegation.values()).reduce(function (
```

Comment:
Existing code in this repo mostly silences errors in a similar way. But this will just throw `TypeError: delegation.values is not a function`

Reply:
Still to handle

Reply:
This has been addressed.